launch_testing_ros/launch_testing_ros/wait_for_topics.py

```diff
@@ -85,6 +92,13 @@ def _prepare_ros_node(self):
 
     def wait(self):
         self.__ros_node.start_subscribers(self.topic_tuples)
+        if self.callback:
+            if isinstance(self.callback_arguments, dict):
+                self.callback(**self.callback_arguments)
+            elif isinstance(self.callback_arguments, (list, set, tuple)):
+                self.callback(*self.callback_arguments)
+            else:
+                self.callback(self.callback_arguments)
         return self.__ros_node.msg_event_object.wait(self.timeout)
```

Comment:
I'm facing a race condition here when CI gets bogged down. 

Seems like when the subscriptions are created in `self.__ros_node.start_subscribers(self.topic_tuples)`, there is no waiting to see that the DDS topics match with the publisher DUT. 

Thus, we are setting to wait on an event handler with a timeout, but the topic isn't matched yet. 

In CI, if we add a `time.sleep(5)`  between `start_subscribers` and `__ros_node.msg_event_object.wait(self.timeout)`, it works fine. 

Is there a way to say:
```python
MATCH_TIMEOUT=5 # How long to wait for all subscribers to match to the publisher topics
now = time.time()
while(not self.__ros_node.all_subscribers_matched() and time.time() - now < MATCH_TIMEOUT ) 
   time.sleep(0.1) # Wait for subscribers to match before proceeding
```

Reply:
Thank you so much for the research. I think I was able to solve the issue by using events as you suggested
177a85aa9b586749bbc90f52d19093cd103d2e9f

Reply:
My team sunk a crazy amount of time into this trying to get it working on humble. Nothing seems to work reliably without the events matching, even counting publishers and subscribers. Because events aren't backported to humble, I have not identified any path forward. 

Do you have interest in finding something for humble too?

Reply:
Yes, sure. Could you please explain how I could reproduce the issue you are facing on humble?

Reply:
I wonder if it would be easier to just backport events to humble too.

Reply:
From what I researched, it would be a heavy lift:
https://github.com/ros2/rmw/issues/330

Reply:
@adityapande-1995 @methylDragon @mjcarroll 
Do you have any advice about this?